new-packages/ts-project/__tests__/source-edits/add-state.test.ts

```diff
@@ -0,0 +1,714 @@
+import { outdent } from 'outdent';
+import { expect, test } from 'vitest';
+import { createTestProject, testdir, ts } from '../utils';
+
+test('should append a state after existing states (with trailing comma)', async () => {
+  const tmpPath = await testdir({
+    'tsconfig.json': JSON.stringify({}),
+    'index.ts': ts`
+      import { createMachine } from "xstate";
+
+      createMachine({
+        states: {
+          foo: {},
+        },
+      });
+    `,
+  });
+
+  const project = await createTestProject(tmpPath);
+
+  const textEdits = project.editDigraph(
+    {
+      fileName: 'index.ts',
+      machineIndex: 0,
+    },
+    [
+      {
+        type: 'add_state',
+        path: [],
+        name: 'bar',
+      },
+    ],
+  );
+  expect(await project.applyTextEdits(textEdits)).toMatchInlineSnapshot(`
+    {
+      "index.ts": "import { createMachine } from "xstate";
+
+    createMachine({s
```

Comment:
Typo?

Reply:
yeah, copy-paste from what I fixed [here](https://github.com/statelyai/xstate-tools/pull/469/commits/3e1c944f1f73428288e5fe90b143e68132b2ac37). My cmd is slightly broken (😭) and that's likely a result of failed `cmd+s` attempt

Reply:
ah, that was just the not updated snapshot after that typo fix